pytorch_lightning/metrics/classification/iou.py

```diff
@@ -0,0 +1,102 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Optional
+import torch
+from pytorch_lightning.metrics.functional.reduction import reduce
+from pytorch_lightning.metrics.functional.confusion_matrix import _confusion_matrix_compute
+from pytorch_lightning.metrics.functional.iou import _iou_from_confmat
+from pytorch_lightning.metrics.classification.confusion_matrix import ConfusionMatrix
+
+
+class IoU(ConfusionMatrix):
+    """
+    Intersection over union, or Jaccard index calculation. Works with binary, multiclass, and multilabel data.
```

Comment:
Can you add link + formula, similar to how other metrics got updated in this PR (#4823). Relevant wiki link is [this](https://en.wikipedia.org/wiki/Jaccard_index).

Reply:
Added. Please take a look.